src/inmanta/env.py

```diff
@@ -469,7 +469,15 @@ def _run_pip_install_command(
             constraints_files=constraints_files,
             requirements_files=requirements_files,
         )
-        return_code, full_output = self.run_command_and_stream_output(cmd)
+
+        # if index_urls are set, only use those. Otherwise, use the one from the environment
+        process_env = os.environ.copy()
```

Comment:
Naming that variable `process_env` is a bit confusing. The `env.py` has a global variable that's also called `process_env`, which refers to the Python environment of this process. It would be good to rename this one.